pkg/apiserver/rest/scan_estimation_controller.go

```diff
@@ -131,6 +162,29 @@ func (s *ServerImpl) PutScanEstimationsScanEstimationID(ctx echo.Context, scanEs
 	}
 	scanEstimation.Id = &scanEstimationID
 
+	// check that a scan estimation with that id exists.
+	existingScanEstimation, err := s.dbHandler.ScanEstimationsTable().GetScanEstimation(scanEstimationID, models.GetScanEstimationsScanEstimationIDParams{})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("scan estimation was not found. scanEstimationID=%v: %v", scanEstimationID, err))
+		}
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get scan estimation. scanEstimationID=%v: %v", scanEstimationID, err))
+	}
+
+	// check for valid state transition
+	status, ok := scanEstimation.GetStatus()
+	if !ok {
+		return sendError(ctx, http.StatusBadRequest, err.Error())
+	}
+	existingStatus, ok := existingScanEstimation.GetStatus()
+	if !ok {
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to retrieve Status for existing scan estimation: scanEstimationID=%v", existingScanEstimation.Id))
+	}
+	err = existingStatus.IsValidTransition(status)
+	if err != nil {
+		return sendError(ctx, http.StatusBadRequest, err.Error())
+	}
+
```

Comment:
we could possibly move this block to a separate function since we repeat it twice just in this file

Reply:
I'd not include it in this PR because it is like this in every controller, in a follow up PR we can change every one of them though.

Reply:
top 😄